Client/Application/AppDelegate.swift

```diff
@@ -81,6 +82,22 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         return true
     }
 
+    func application(application: UIApplication, openURL url: NSURL, sourceApplication: String?, annotation: AnyObject?) -> Bool {
+        if url.scheme == "firefox" {
+            if url.lastPathComponent == "open-url" {
+                if let query = url.query {
+                    if let index = query.unescape().rangeOfString("=") {
+                        let newRequest = NSURLRequest(URL: NSURL(string: query.substringFromIndex(index.endIndex))!)
```

Comment:
The forced unwrapping of `NSURL(string: ...)` is problematic. If another app sends a bad URL, then `NSURL(string:)` will return `nil` and that will cause a crash when it is passed to `NSURLRequest(URL:)`. Every time you force unwrap (using `!`) you have to ask yourself, can this operation fail or am I 100% sure that it is safe. If you are not sure then use a conditional like `if let` to make the code more robust.